src/shared/createActivity.ts

```diff
@@ -0,0 +1,62 @@
+import { Activity, User } from "../../generated/schema";
+import { BigInt, ethereum, Address } from "@graphprotocol/graph-ts";
+import { getOrCreateUser } from './getOrCreateUser';
+// import { log } from '@graphprotocol/graph-ts'
+
+// enum ActivityType {
+//   Vote
+//   Stake
+//   Unstake
+//   Ratify
+//   Reject
+//   Create
+//   Veto
+// }
+
+export function getActivityId(
+  activityType: string,
+  userAddress: string,
+  transactionHash: string,
+  logIndex: string
+): string {
+  return "activity-" + activityType + "-" + userAddress + "-" + transactionHash + "-" + logIndex;
```

Comment:
is this too long ?

Reply:
I’m not sure i need the logIndex. can a user perform the same activity on the same transaction multiple times? 
Also is logIndex appropriate or should i use nounce?

Reply:
The user can technically perform the same activity on the same transaction multiple times. What you can do is hash all of the fields together to make an identifier so it's more compact. LogIndex is necessary.

Reply:
Okay I kept "activity" and activityType as a prefix and hashed the rest. I am actually using a substring of the hash that is 24 chars long. The p(collision) is negligible